src/Microsoft.Data.Analysis/PrimitiveColumnContainer.cs

```diff
@@ -251,6 +251,52 @@ public void ApplyElementwise(Func<T?, long, T?> func)
             }
         }
 
+        /// <summary>
+        /// Applies the given <paramref name="func"/> over a window of size <paramref name="windowSize"/> of the column's values
+        /// </summary>
+        /// <typeparam name="TResult">The return value for each window</typeparam>
+        /// <param name="func">The delegate to apply over each window. It takes an input of the values in the current window and the current index being processed in the column. </param>
+        /// <param name="resultContainer">The result container</param>
+        /// <param name="windowSize">Size of the window</param>
+        public void Apply<TResult>(Func<LinkedList<T?>, long, TResult?> func, PrimitiveColumnContainer<TResult> resultContainer, int windowSize)
+            where TResult : unmanaged
+        {
+            LinkedList<T?> list = new LinkedList<T?>();
+            for (int b = 0; b < Buffers.Count; b++)
+            {
+                ReadOnlyDataFrameBuffer<T> buffer = Buffers[b];
+                long prevLength = checked(Buffers[0].Length * b);
+                DataFrameBuffer<T> mutableBuffer = DataFrameBuffer<T>.GetMutableBuffer(buffer);
+                Buffers[b] = mutableBuffer;
```

Comment:
Why are we setting anything here? Or making a mutableBuffer on the current column container? This shouldn't be modifying the current container at all - it should just be modifying the `resultContainer`.